File: Python/5073.py
while 1:
    S = input().split(" ")
    if S[0] == '0' and S[1] == '0' and S[2] == '0':
        break
    else:
        S.sort()
        a,b,c = int(S[0]) , int(S[1]), int(S[2])
        if c >= a + b:
            print('Invalid')
        else :
            if a == b and b == c and a == c:
                print('Equilateral')
            elif (a == b and b != c) or (a != b and b == c):
                print('Isosceles')
            elif a != b and b != c and a != c :
                print('Scalene')

File: C_C++/5073.cpp
#include <algorithm>
#include <stdio.h>

using namespace std;

int inp[4];

int main() {
  while(1){
    scanf("%d %d %d", &inp[0], &inp[1], &inp[2]);
    if (inp[0] == 0 && inp[1] == 0 && inp[2] == 0) break;
    sort(inp, inp+3);
    if (inp[2] >= inp[0] + inp[1]) printf("Invalid\n");
    else if (inp[0] == inp[1] && inp[1] == inp[2]) printf("%s\n", "Equilateral");
    else if (inp[0] == inp[1] || inp[1] == inp[2]) printf("%s\n", "Isosceles");
    else printf("%s\n", "Scalene");
  }
}
